### PR TITLE
Add facet option (solr missing)

### PR DESCRIPTION
**Pull request by chrizzor**
This PR extends the facet configuration.
With "showmissing = 1" it is now possible to show all matching documents with no facet value.
The label for this facet option can be configured with "labelmissing".

### CODE REVIEW

File: Classes/Service/SolrServiceProvider.php [hunk 1]
```diff
@@ -301,14 +301,21 @@ protected function addFacetFilters(array $arguments): array
                         $queryInfo['tag'] = $this->tagForFacet($facetID);
                     }
 
-                    $this->query->createFilterQuery($queryInfo)
-                        ->setQuery($facetQuery);
+                    // If facet.missing is active and facet is selected
+                    // set solr query to exclude all known facet values
+                    if ($facetTerm == $facetInfo['config']['labelmissing']) {
```
Comment (ipf) [inline Classes/Service/SolrServiceProvider.php:306]: 1. Please add the documentation / example for this parameter
2. For this parameter, please use a camelCasedVersion - `labelMissing`
3. Please use strict comparison `===`

File: Classes/Service/SolrServiceProvider.php [hunk 2]
```diff
@@ -357,6 +364,10 @@ protected function addFacetQueries(): void
                     if (1 == $facet['excludeOwnFilter']) {
                         $queryForFacet->addExclude($this->tagForFacet($facetID));
                     }
+
+                    if ($facet['showmissing'] == 1) {
```
Comment (ipf) [inline Classes/Service/SolrServiceProvider.php:368]: Please use strict comparison. I know, the code part above does not use it, but we're about to change the strictness of the extension.

**Review by ipf — CHANGES_REQUESTED** [review]

**Comment by ipf** [comment]
Please rebase, fix the style with `composer lint` and check, why there are new indents in the Readme.

**Comment by ipf** [comment]
Please rebase, fix the style with `composer lint` and check, why there are new indents in the Readme.

**Comment by chrizzor** [comment]
Rebased and styling fixed

**Pull request merged**